api/pkg/app/app.go

```diff
@@ -168,17 +171,47 @@ func (ab *APIBase) Data() *Data {
 
 // ReloadData reads config file and loads data in Data object
 func (ab *APIBase) ReloadData() error {
-	// Reads config file url from env
-	url, err := configFileURL()
-	if err != nil {
-		return err
-	}
 
-	// Reads data from config file
-	fileData, err := dataFromURL(url)
-	if err != nil {
-		ab.logger.Errorf("failed to read config file: %v", err)
-		return err
+	var fileData []byte
+
+	if ab.Environment() == Production {
+		// checks Config dir is present or not
+		dir, err := os.ReadDir(ConfigFileDir)
+		if err != nil {
+			ab.logger.Errorf("Directory %s not found : %s", ConfigFileDir, err)
+			return err
+		}
+		// Iterate on all keys to construct file of config data by adding key,value pair
+		for _, f := range dir {
+			//Skips to read hiden files
+			if !strings.Contains(f.Name(), "..") {
+				d, err := readLocalFile(ConfigFileDir + f.Name())
+				if err != nil {
+					return err
+				}
+				if strings.ToLower(f.Name()) == CatalogRefreshInterval {
+					continue
+				} else if strings.ToLower(f.Name()) == DefaultScopes {
+					fileData = append(fileData, []byte(fmt.Sprintf("\n%s:\n %s", strings.ToLower(f.Name()), string(d)))...)
+				} else {
+					fileData = append(fileData, []byte(fmt.Sprintf("\n%s:\n%s", strings.ToLower(f.Name()), string(d)))...)
+				}
+			}
```

Comment:
In both the condition and else part we are adding the same logic, can we then skip the else if part then ?

Reply:
in the else part it adds an extra white space